inbox/helpers_test.go

```diff
@@ -25,26 +27,96 @@ func TestGetGroupAndAssociatedMemberIdentifier(t *testing.T) {
 		},
 	}
 
-	results, err := getGroupAndAssociatedMemberIdentifier(jamesIdentifier, nil)
+	results, err := GetGroupAndAssociatedMemberIdentifier(jamesIdentifier)
 	if err != nil {
 		t.Error(err)
 		return
 	}
 
-	if len(results) != 2 {
-		t.Error(fmt.Errorf("length of result doesn't match: expected: %v, got: %v", 3, len(results)))
-		return
+	expected := []GroupAndAssociatedMemberIdentifier{
+		{
+			Group: ObjectIdentifier{
+				ObjectName:     "database",
+				ObjectUniqueID: "1",
+				ObjectType:     7,
+				IsGroupType:    true,
+				ParentObject: &ObjectIdentifier{
+					ObjectName:     "namespace",
+					ObjectUniqueID: "2",
+					ObjectType:     7,
+					IsGroupType:    true,
+					ParentObject: &ObjectIdentifier{
+						ObjectName:     "cluster",
+						ObjectUniqueID: "3",
+						ObjectType:     7,
+						IsGroupType:    true,
+					},
+				},
+			},
+			Member: ObjectIdentifier{
+				ObjectName:     "namespace",
+				ObjectUniqueID: "2",
+				ObjectType:     7,
+				IsGroupType:    true,
+				ParentObject: &ObjectIdentifier{
+					ObjectName:     "cluster",
+					ObjectUniqueID: "3",
+					ObjectType:     7,
+					IsGroupType:    true,
+				},
+			},
+		},
+		{
+			Group: ObjectIdentifier{
+				ObjectName:     "namespace",
+				ObjectUniqueID: "2",
+				ObjectType:     7,
+				IsGroupType:    true,
+				ParentObject: &ObjectIdentifier{
+					ObjectName:     "cluster",
+					ObjectUniqueID: "3",
+					ObjectType:     7,
+					IsGroupType:    true,
+				},
+			},
+			Member: ObjectIdentifier{
+				ObjectName:     "cluster",
+				ObjectUniqueID: "3",
+				ObjectType:     7,
+				IsGroupType:    true,
+			},
+		},
+	}
+
+	if err = isEqual(results, expected); err != nil {
+		t.Error(err)
 	}
+}
 
-	for _, result := range results {
-		fmt.Printf("group: %s, member: %s\n", result.Group.ObjectName, result.Member.ObjectName)
-		if !result.Member.HasParentObject() {
-			t.Errorf("member %s does not have parent object", result.Member.ObjectName)
-			return
+func isEqual(results, expected []GroupAndAssociatedMemberIdentifier) error {
```

Comment:
isEqual is intended to return true or false. you can name it compare and return error